### ISSUE TITLE
Can anyone help me?

### ISSUE THREAD

**Issue by liliang198909** [gensyn-ai/rl-swarm#303]
![Image](https://github.com/user-attachments/assets/b79c0b20-ea75-4e18-ba6d-8a017ef110d6)

 85%|████████████████████████████████████████████████████████████████▌           | 17/20 [13:46<03:05, 61.82s/it]./run_rl_swarm.sh: line 266: 33498 Killed: 9               python -m hivemind_exp.gsm8k.train_single_gpu --hf_token "$HUGGINGFACE_ACCESS_TOKEN" --identity_path "$IDENTITY_PATH" --modal_org_id "$ORG_ID" --contract_address "$SWARM_CONTRACT" --config "$CONFIG_PATH" --game "$GAME"
>> Shutting down trainer...
zsh: terminated  ./run_rl_swarm.sh
(venv) lee@leedeMac-mini rl-swarm % /opt/homebrew/Cellar/python@3.13/3.13.3_1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py:301: UserWarning: resource_tracker: There appear to be 5 leaked semaphore objects to clean up at shutdown: {'/mp-wl6oawpc', '/mp-b0g1svfu', '/mp-puqkp036', '/mp-8otw36w2', '/mp-9q5hkgar'}
  warnings.warn(

**Comment by liliang198909** [comment]
How many have you run?

**Comment by liliang198909** [comment]
how did you solve it ?



---Original---
From: ***@***.***&gt;
Date: Mon, May 26, 2025 00:16 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [gensyn-ai/rl-swarm] Can anyone help me? (Issue #303)


DonS598 left a comment (gensyn-ai/rl-swarm#303)
 
Only one time I had such a problem.

**Comment by jitkumark1232** [comment]
It's due to the Ram memory shortage issue, u might be needed to lower the below parameters

for cpu

#Model arguments
model_revision: main
torch_dtype: float32
bf16: false
tf32: false2

#Training arguments
gradient_accumulation_steps: 4

#GRPO arguments
per_device_train_batch_size: 4

set gradient_accumulation_steps: & gradient_accumulation_steps: 4 to 2 and try restarting the node.

**Comment by liliang198909** [comment]
# Model arguments
model_revision: main
torch_dtype: bfloat32
bf16: false
tf32: false


# Dataset arguments
dataset_id_or_path: 'openai/gsm8k'


# Training arguments
max_steps: 20 # Original 450
gradient_accumulation_steps: 4
gradient_checkpointing: true
gradient_checkpointing_kwargs:
&nbsp; use_reentrant: false
learning_rate: 5.0e-7 # 1.0e-6 as in the deepseek math paper 5-e7 from https://hijkzzz.notion.si$
lr_scheduler_type: cosine
warmup_ratio: 0.03


# GRPO arguments
use_vllm: false
num_generations: 2
per_device_train_batch_size: 1
beta: 0.001 # 0.04 as in the deepseek math paper 0.001 from https://hijkzzz.notion.site/unraveli$
max_prompt_length: 224
max_completion_length: 448


# Logging arguments
logging_strategy: steps&nbsp; &nbsp;
logging_steps: 2
report_to:
- tensorboard
save_strategy: "steps"
save_steps: 25
seed: 42


# Script arguments
max_rounds: 10000


# Model-specific arguments
model_name_or_path: unsloth/Qwen2.5-0.5B-Instruct
output_dir: runs/gsm8k/multinode/Qwen2.5-0.5B-Instruct-Gensyn-Swarm




---Original---
From: ***@***.***&gt;
Date: Mon, May 26, 2025 02:32 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [gensyn-ai/rl-swarm] Can anyone help me? (Issue #303)


jitkumark1232 left a comment (gensyn-ai/rl-swarm#303)
 
It's due to the Ram memory shortage issue, u might be needed to lower the below parameters
 
for cpu
 
#Model arguments
 model_revision: main
 torch_dtype: float32
 bf16: false
 tf32: false2
 
#Training arguments
 gradient_accumulation_steps: 4
 
#GRPO arguments
 per_device_train_batch_size: 4
 
set gradient_accumulation_steps: &amp; gradient_accumulation_steps: 4 to 2 and try restarting the node.

**Comment by liliang198909** [comment]
What device do you use? I use an MAC  m4.

**Comment by liliang198909** [comment]
What are your other parameters?

**Comment by jitkumark1232** [comment]
hey, bro just apply the change that i recommended to u, although if u running it on ur own system make sure u have sufficient ram.

I' using same parameters as urs but I have applied some changes that I mentioned earlier
